### PR TITLE
naming rules clarification

### PR DESCRIPTION
**Pull request by AlexicaWright**

### CODE REVIEW

**Review by craigtaverner — APPROVED** [review]
I think this is an improvement, although I preferred the idea of making the escaping of strings something written once right at the top instead of repeated as an extra bullet point on every single part.

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 1]
```diff
@@ -5,22 +5,30 @@ We describe here rules and recommendations for the naming of node labels, relati
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc:5]: I know it's something you haven't changed but still I would remove the 'we' form from the page and make it descriptive instead of personal. Same for the last sentence (our recommendations).

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 2]
```diff
@@ -5,22 +5,30 @@ We describe here rules and recommendations for the naming of node labels, relati
 
 == Naming rules
 
-* Must begin with an alphabetic letter.
+* Alphabetic letters:
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc]: Isn't this a tautology? Letters can only be alphabetic. I would rather rephrase it to either:
Alphabetic characters or just Letters.

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 3]
```diff
@@ -5,22 +5,30 @@ We describe here rules and recommendations for the naming of node labels, relati
...
+** Names should begin with an alphabetic letter.
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc]: same as above

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc (hunk 1 above)
Comment (renetapopova) [reply cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc:5]: Something like:
"This section describes the set of rules and recommendations to follow when naming node labels, relationship types, property names, and variables."

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 4]
```diff
@@ -5,22 +5,30 @@ We describe here rules and recommendations for the naming of node labels, relati
...
+** Can be very long, up to `65535` (`2^16 - 1`) or `65534` characters, depending on the version of Neo4j.
+* Case-sensitive:
 ** Thus, `:PERSON`, `:Person` and `:person` are three different labels, and `n` and `N` are two different variables.
 * Whitespace characters:
 ** Leading and trailing whitespace characters will be removed automatically. For example, `MATCH (  a  ) RETURN a` is equivalent to `MATCH (a) RETURN a`.
-** If spaces are required within a name, use backticks for escaping; e.g. `++`my variable has spaces`++`.
+//** If spaces are required within a name, use backticks for escaping; e.g. `++`my variable has spaces`++`.
+
+[NOTE]
+Non-alphabetic letters, numbers, symbols and whitespace characters *can* be used in names, but *must* be escaped using backticks.
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc]: What does a non-alphabetic letter mean?

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 5]
```diff
@@ -5,22 +5,30 @@ We describe here rules and recommendations for the naming of node labels, relati
...
 
-* Must begin with an alphabetic letter.
+* Alphabetic letters:
+** Names should begin with an alphabetic letter.
 ** This includes "non-English" characters, such as `å`, `ä`, `ö`, `ü` etc.
-** If a leading non-alphabetic character is required, use backticks for escaping; e.g. `++`^n`++`.
-* Can contain numbers, but not as the first character.
+//** If a leading non-alphabetic character is required, use backticks for escaping; e.g. `++`^n`++`.
+* Numbers:
+** Names should not begin with a number without escaping.
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc]: I would rather rewrite it as you suggested, first what is possible and then what should be avoided, and if you can't how to tackle it.

Comment (renetapopova) [reply cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc]: For example, "Names can contain numbers but not as the first character, for example, `first1`.

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 6]
```diff
@@ -5,22 +5,30 @@ We describe here rules and recommendations for the naming of node labels, relati
...
 ** To illustrate, `1first` is not allowed, whereas `first1` is allowed.
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc:13]: If the above is accepted, then this sentence is not needed.

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 7]
```diff
@@ -5,22 +5,30 @@ We describe here rules and recommendations for the naming of node labels, relati
...
-** If a leading numeric character is required, use backticks for escaping; e.g. `++`1first`++`.
-* Cannot contain symbols.
+//** If a leading numeric character is required, use backticks for escaping; e.g. `++`1first`++`.
+* Symbols:
+** Names cannot contain symbols unless these are escaped.
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc]: Names cannot contain symbols except for underscore, e.g, `my_variable`, and `$` as the first character to denote a parameter, as given by `$myParam`.

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 8]
```diff
@@ -5,22 +5,30 @@ We describe here rules and recommendations for the naming of node labels, relati
...
 ** An exception to this rule is using underscore, as given by `my_variable`.
 ** An ostensible exception to this rule is using `$` as the first character to denote a <<cypher-parameters, parameter>>, as given by `$myParam`.
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc]: If the above is accepted, then these sentences are not needed.

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 9]
```diff
@@ -5,22 +5,30 @@ We describe here rules and recommendations for the naming of node labels, relati
...
-** If a leading symbolic character is required, use backticks for escaping; e.g. `++`$$n`++`.
-* Can be very long, up to `65535` (`2^16 - 1`) or `65534` characters, depending on the version of Neo4j.
-* Are case-sensitive.
+//** If a leading symbolic character is required, use backticks for escaping; e.g. `++`$$n`++`.
+* Length:
+** Can be very long, up to `65535` (`2^16 - 1`) or `65534` characters, depending on the version of Neo4j.
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc:17]: I would start from "Up to ..."

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 10]
```diff
@@ -5,22 +5,30 @@ We describe here rules and recommendations for the naming of node labels, relati
...
+* Case-sensitive:
 ** Thus, `:PERSON`, `:Person` and `:person` are three different labels, and `n` and `N` are two different variables.
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc]: I would add one bullet before this one:
"Names are case-sensitive."

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 11]
```diff
@@ -5,22 +5,30 @@ We describe here rules and recommendations for the naming of node labels, relati
...
 * Whitespace characters:
 ** Leading and trailing whitespace characters will be removed automatically. For example, `MATCH (  a  ) RETURN a` is equivalent to `MATCH (a) RETURN a`.
-** If spaces are required within a name, use backticks for escaping; e.g. `++`my variable has spaces`++`.
+//** If spaces are required within a name, use backticks for escaping; e.g. `++`my variable has spaces`++`.
+
+[NOTE]
+Non-alphabetic letters, numbers, symbols and whitespace characters *can* be used in names, but *must* be escaped using backticks.
+For example: `++`^n`++`, `++`1first`++`, `++`$$n`++`, and `++`my variable has spaces`++`.
 
 == Scoping and namespace rules
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc:29]: The last "etc" is missing a .

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc (hunk 4 above)
Comment (renetapopova) [reply cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc]: My suggestion: "Numbers, symbols and whitespace characters *can be used in names only if escaped by backticks*.

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc [hunk 12]
```diff
@@ -1,26 +1,35 @@
...
+** Names are case-sensitive and thus, `:PERSON`, `:Person` and `:person` are three different labels, and `n` and `N` are two different variables.
 * Whitespace characters:
-** Leading and trailing whitespace characters will be removed automatically. For example, `MATCH (  a  ) RETURN a` is equivalent to `MATCH (a) RETURN a`.
-** If spaces are required within a name, use backticks for escaping; e.g. `++`my variable has spaces`++`.
+** Leading and trailing whitespace characters will be removed automatically.
+For example, `MATCH (  a  ) RETURN a` is equivalent to `MATCH (a) RETURN a`.
+//** If spaces are required within a name, use backticks for escaping; e.g. `++`my variable has spaces`++`.
+
+[NOTE]
+Non-alphabetic characters, including numbers, symbols and whitespace characters, *can* be used in names, but *must* be escaped using backticks.
```
Comment (craigtaverner) [inline cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc:25]: I now actually like this version with the exceptions all grouped at the end. Makes the normal part above easier to read and cleaner.

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc (hunk 1 above)
Comment (AlexicaWright) [reply cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc:5]: Agree!

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc (hunk 9 above)
Comment (AlexicaWright) [reply cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc:17]: I think it is worth pointing out that they indeed can be very long.

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc (hunk 12 above)
Comment (AlexicaWright) [reply cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc:25]: Thank you @craigtaverner , I'll keep it this way then!

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc (hunk 2 above)
Comment (AlexicaWright) [reply cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc]: Thank you @renetapopova , you are absolutely right! Addressed in all instances.

File: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc (hunk 11 above)
Comment (AlexicaWright) [reply cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc:29]: Found it and added it!
